editoast/src/modelsv2/infra.rs

```diff
@@ -243,6 +245,22 @@ impl Infra {
         let voltages = sql_query(query).load::<Voltage>(conn).await?;
         Ok(voltages)
     }
+
+    pub async fn get_speed_limit_tags(
+        conn: &mut DbConnection,
+        infra_id: i64,
+    ) -> Result<Vec<SpeedLimitTags>> {
```

Comment:
```suggestion
    pub async fn get_speed_limit_tags(
        &self,
        conn: &mut DbConnection,
    ) -> Result<Vec<SpeedLimitTags>> {
```

Reply:
```
#[get("/speed_limit_tags")]
async fn get_speed_limit_tags(
    infra: Path<InfraIdParam>,
    db_pool: Data<DbConnectionPool>,
) -> Result<Json<Vec<String>>> {
    let conn = &mut db_pool.get().await?;
    let speed_limits_tags = Infra::get_speed_limit_tags(conn, infra.infra_id).await?;
    Ok(Json(
        speed_limits_tags.into_iter().map(|el| (el.tag)).collect(),
    ))
}
```
we don't an instance of infra during the request

Reply:
Then we should `retrieve_of_fail` it. Because of that, right now we don't return an error if the `infra_id` is invalid, and it should be the case. `retrieve`-ing it solves that.

Reply:
Done.